### PR TITLE
Adds UInt64 parsing for raRef:CResource in TweakDB compilation

### PR DESCRIPTION
**Pull request by jackhumbert**
Only parses decimal since that's the only format I've ever seen CResources in - please correct me if I'm mistaken!

### CODE REVIEW

File: WolvenKit.RED4.TweakDB/Types/CResource.cs [hunk 1]
```diff
@@ -11,11 +10,21 @@ public class CResource : IPrimitive
...
 
-        public static implicit operator CResource(string value) => new()
+        public static implicit operator CResource(string value)
         {
-            Text = value,
-            Key = (!string.IsNullOrEmpty(value) && value != "0") ? FNV1A64HashAlgorithm.HashString(value) : 0 
-        };
+            var cres = new CResource();
+            UInt64 key;
+            if (UInt64.TryParse(value, result: out key))
```
Comment (WopsS) [inline WolvenKit.RED4.TweakDB/Types/CResource.cs:17]: `ulong.TryParse(value, out var key)` and the declaration above is not needed.

File: WolvenKit.RED4.TweakDB/Types/CResource.cs [hunk 2]
```diff
@@ -11,11 +10,21 @@ public class CResource : IPrimitive
...
+            {
+                cres.Text = value;
+                cres.Key = key;
+            } else
```
Comment (WopsS) [inline WolvenKit.RED4.TweakDB/Types/CResource.cs:21]: `else` on new line.

**Review by WopsS — CHANGES_REQUESTED** [review]

File: WolvenKit.RED4.TweakDB/Types/CResource.cs (hunk 1 above)
Comment (jackhumbert) [inline WolvenKit.RED4.TweakDB/Types/CResource.cs:17]:
```suggestion
            if (ulong.TryParse(value, out var key))
```

File: WolvenKit.RED4.TweakDB/Types/CResource.cs (hunk 2 above)
Comment (jackhumbert) [inline WolvenKit.RED4.TweakDB/Types/CResource.cs:21]:
```suggestion
            } 
            else
```

**Review by WopsS — APPROVED** [review]

**Pull request merged**